package NYT.NChunkClient.NProto;

import "yt/core/misc/guid.proto";
import "yt/core/misc/protobuf_helpers.proto";
import "yt/ytlib/chunk_client/schema.proto";

////////////////////////////////////////////////////////////////////////////////

// Describes a portion of table chunk.
message TChunkSpec
{
    required NYT.NProto.TGuid chunk_id = 1;

    // Lower boundary, inclusive.
    optional TReadLimit start_limit = 2;

    // Upper boundary, exclusive.
    optional TReadLimit end_limit = 3;

    // Initial seed addresses. The reader may ask the master for additional ones.
    repeated fixed32 replicas = 4;

    // Column filter.
    optional TChannel channel = 5;

    required NYT.NProto.TExtensionSet extensions = 6;

    optional int32 table_index = 7 [default = -1];

    optional int32 partition_tag = 8 [default = -1];

    optional int32 erasure_codec = 9 [default = 0];

    optional int64 table_row_index = 10 [default = 0];
}

////////////////////////////////////////////////////////////////////////////////
